### ISSUE TITLE
TC Inaccurate Temperature

### ISSUE THREAD

**Issue by flavius-t** [EdgePi-Cloud/edgepi-python-sdk#71]
The thermocouple is measuring temperatures that are higher than expected. As as example, the following is obtained at room temperature:

![image](https://user-images.githubusercontent.com/77416463/179045207-37026196-dd98-4ad7-bbcc-c5e9544f5907.png)

The issue is not with temperature code conversion, as tests confirm these are the actual temperature values being stored in the thermocouple chip's registers.

We need to understand what factors are causing this issue:
- maybe a cold-junction offset value needs to be applied

**Comment by farzadpanahi** [comment]
@sjpark608 I think this is the same issue @jarell-santella  is seeing. can you confirm Jarell?

**Comment by jarell-santella** [comment]
> @sjpark608 I think this is the same issue @jarell-santella is seeing. can you confirm Jarell?

I think this is exactly the same issue. Same reading too. ~28 degrees celsius is being read at room temperature with probe type K. I have tried disabling cold junction temperature readings, but it doesn't make a difference.

**Comment by sjpark608** [comment]
hmm, i thought it might be related to the Thermocouple we are using on the unit. I will try a brand-new thermocouple.

**Comment by sjpark608** [comment]
The following section of the datasheet explains the effect of the series resistance in the temperature measurement due to the voltage difference introduced by these resistors. Opening a tech support case for some clarification.
![image](https://user-images.githubusercontent.com/78987042/212759795-ec0ec766-fb0d-43c0-b837-74269f9e7fa1.png)

**Comment by farzadpanahi** [comment]
@sjpark608 what need to be done on our side? do we need some sort of calibration for TC?

**Comment by sjpark608** [comment]
> @sjpark608 what need to be done on our side? do we need some sort of calibration for TC?

yes, we can gather some measurements at different temperature and add the calibration value.
